scripts/filter_multimappers.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+
+"""Filter multimappers in a SAM file by indel count.
+
+This script uses the pysam library to open the input SAM file and iterates
+over each alignment in it. It is assumed that the file is ordered by query
+name. For each alignment, it checks if it is a secondary or supplementary
+alignment, if it is, it skips to the next alignment. Otherwise, it adds the
+alignment to a list storing all alignments with the same query name.
+If the current query changes, it selects the best alignment(s) from the list
+of current alingments and writes them to the standard output. Current query
+and current alignments list are reset to the new query and alignment
+respectively.
+Finally, if there are any remaining alignments in the current alignments
+list and selects the best alignment(s) for those queries, writing them to
+standard output.
+
+Input:
+    input_sam_file (str): Path to the input SAM file, ordered by query name.
+
+Output:
+    standard output: Filtered SAM file
+
+Functions:
+    parse_arguments():
+        Command-line arguments parser
+    count_indels(pysam.AlignedSegment):
+        Count the number of indels in an alignment based on its CIGAR string.
+    find_best_alignments(list[pysam.AlignedSegment]):
+        Find alignments with less indels
+    write_output(list[pysam.AlignedSegment]):
+        Write the output to the standard output (stdout).
+    main(Path):
+        Filter multimappers by indels count.
+
+Usage:
+    python filter_multimappers.py -i input_sam_file
+    python filter_multimappers.py -i input_sam_file > output_sam_file
+"""
+from pathlib import Path
+from typing import List
+import argparse
+import pysam
+import sys
+
+
+def parse_arguments():
+    """Command-line arguments parser."""
+    parser = argparse.ArgumentParser(
+        description="Script to filter multimappers by indel counts."
+        )
+
+    parser.add_argument(
+        '-v', '--version',
+        action='version',
+        version='%(prog)s 1.0',
+        help="Show program's version number and exit"
+        )
+
+    parser.add_argument(
+        '-i', '--infile',
+        help="Path to the SAM input file, sorted by query name.",
+        type=Path
+        )
+
+    return parser
+
+
+def count_indels(alignment: pysam.AlignedSegment) -> int:
+    """Count the number of indels in an alignment based on its CIGAR string.
+
+    This function counts the number of indels in the alignment based on the
+    alignment CIGAR string returned by .cigartuples. Insertions are encoded as
+    1s whereas the deletions are encoded as 2.
+    Refer to the .cigartuples documentation for more information at:
+    https://pysam.readthedocs.io/en/v0.15.0/api.html#pysam.AlignedSegment.cigartuples
+
+    Args:
+        alignments(pysam.AlignedSegment): The read to count insertions for.
```

Comment:
No types in the docstrings please. We always want to avoid situations where we have multiple sources of truth. Otherwise, the risk is too high that one, but not the other part gets updated. If possible, it's always better to have the code be the single source of truth. For the same reason, we want to have as few inline comments as possible. Use them only in really hard to understand parts of your code. And if you _have_ hard to understand parts in your code, _first_ think whether you can make them easier to understand. And _only_ if that is not possible (or would be much more costly), then add an inline comment.